File: backend/identity-service/src/main/java/com/cine/social/identity/service/AuthService.java
package com.cine.social.identity.service;


import com.cine.social.identity.dto.request.AuthenticationRequest;
import com.cine.social.identity.dto.request.UserCreationRequest;
import com.cine.social.identity.dto.response.AuthenticationResponse;
import com.cine.social.identity.dto.response.UserResponse;

public interface AuthService {
    AuthenticationResponse authenticate(AuthenticationRequest request);
    UserResponse register(UserCreationRequest request);

    AuthenticationResponse outboundAuthenticate(String code);
}

File: frontend/src/features/auth/services/authService.ts
import api from '@/lib/axios';
import type { AuthenticationRequest, UserCreationRequest, ApiResponseAuthenticationResponse, ApiResponseUserResponse } from '../types';

export const authService = {
    login: async (credentials: AuthenticationRequest) => {
        const response = await api.post<ApiResponseAuthenticationResponse>('/identity/api/v1/auth/token', credentials);
        return response.data;
    },
    register: async (data: UserCreationRequest) => {
        const response = await api.post<ApiResponseUserResponse>('/identity/api/v1/auth/register', data);
        return response.data;
    },
    logout: async () => {
        // Optional: Call API to invalidate token if backend requires it
        // For JWT usually client-side removal is enough unless using blacklist
    },
    outboundAuthentication: async (code: string) => {
        const response = await api.post<ApiResponseAuthenticationResponse>('/identity/api/v1/auth/outbound/authentication', null, {
            params: { code }
        });
        return response.data;
    }
};
